packages/flutter_test/lib/src/binding.dart

```diff
@@ -1424,8 +1444,11 @@ class LiveTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     switch (source) {
       case TestBindingEventSource.test:
         if (!renderView._pointers.containsKey(event.pointer)) {
-          assert(event.down);
-          renderView._pointers[event.pointer] = _LiveTestPointerRecord(event.pointer, event.position);
+          assert(event.down || event is PointerAddedEvent);
+          if (event.down) {
```

Comment:
@liyuqian @dnfield This one is one of the place I have to modify framework. And I actually are considering it a bug: if there is a customized Widget that's specifically reacting to `PointerAddedEvent` in some way, because of this, such behavior won't be able to be tested using `flutter_test`.

Reply:
As a work around in my specific project, I can skip all recorded `PointerAddedEvent`.

Reply:
Cool - it's fine if it looks weird, as long as it's valid :)